src/main/java/com/walking/intensive/chapter1/task2/Task2.java

```diff
@@ -33,12 +33,52 @@
  */
 public class Task2 {
     public static void main(String[] args) {
-//        Для собственных проверок можете делать любые изменения в этом методе
+        for (int i = -10; i <= 100; i++) {
+            System.out.println(getFlatLocation(5, 4, i));
+        }
     }
 
     static String getFlatLocation(int floorAmount, int entranceAmount, int flatNumber) {
-        //        Место для вашего кода
+        String flatLocation = "0";
+        String incorrectInputData = "Некорректные входные данные";
+        String unknownFlat = "Такой квартиры не существует";
+        int currentFloor;
+        int currentEntrance;
 
-        return null; // Заглушка. При реализации - удалить
+        //Проверка входных данных
+        if (floorAmount <= 0 || entranceAmount <= 0 || flatNumber <= 0) {
+            return incorrectInputData;
+        }
+        //Проверка вхождения номера квартиры в диапазон "существующих"
+        if (flatNumber > 4 * floorAmount * entranceAmount) {
+            return unknownFlat;
+        }
+
+        //Текущий подъезд
+        currentEntrance = flatNumber % (floorAmount * 4) == 0 ?
+                flatNumber / (floorAmount * 4) :
```

Comment:
Error: eckstyle] [ERROR] /home/runner/work/intensive-tasks-2024/intensive-tasks-2024/src/main/java/com/walking/intensive/chapter1/task2/Task2.java:59:48: ':' should be on a new line. [OperatorWrap]